ca_tools/scripts/control_node.py

```diff
@@ -0,0 +1,185 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2013 PAL Robotics SL.
+# Released under the BSD License.
+#
+# Authors:
+#   * Siegfried-A. Gevatter
+
+import curses
+import math
+from enum import Enum
+import signal
+import sys
+
+import rospy
+from geometry_msgs.msg import Twist
+from geometry_msgs.msg import Pose2D
+from geometry_msgs.msg import Point
+from nav_msgs.msg import Odometry
+import tf
+
+TURNING=1
+STOP_TURNING=2
+FORWARD=3
+STOP_FORWARD=4
+
+def signal_handler(signal, frame,queue_size=1):
+  # your code here
+    pubtopic = "/create1/cmd_vel"
+    pub = rospy.Publisher(pubtopic, Twist,queue_size=1)
+    aux = Twist()
+    aux.angular.x=0.0
+    aux.angular.y=0.0
+    aux.angular.z=0.0
+    aux.linear.x=0.0
+    aux.linear.y=0.0
+    aux.linear.z=0.0
+    pub.publish(aux)
+    sys.exit(0)
+
+class ctrl_Node:
+    
+    def __init__(self):
+        
+        rospy.init_node('control_node')
+        pubtopic = "/create1/cmd_vel"
+        self.my_pub = rospy.Publisher(pubtopic, Twist, queue_size=10)
+        subtopic = "/create1/gts"
+        # Create a subscriber with appropriate topic, custom message and name of callback function.
+        self.my_sub=rospy.Subscriber(subtopic,Odometry,self._callback)
+        self.my_goals = []
+        
+        for i in range(4):
+            self.my_goals.append(Pose2D())
+
+
+        self.my_goals[0].x=0.0
+        self.my_goals[0].y=0.0
+
+        self.my_goals[1].x=2.0
+        self.my_goals[1].y=0.0
+
+        self.my_goals[2].x=2.0
+        self.my_goals[2].y=2.0
+
+        self.my_goals[3].x=0.0
+        self.my_goals[3].y=2.0
+        
+        self.goal_num=0
+        self.state=STOP_FORWARD
+        self.my_pose=Pose2D()
+        self.my_pose.x="unset"
+        self.my_pose.y="unset"
+
+# Create a callback function for the subscriber.
+    def _callback(self,data):
+        self.my_pose.x = data.pose.pose.position.x
+        self.my_pose.y = data.pose.pose.position.y
+        q0 = data.pose.pose.orientation.w
+        q1 = data.pose.pose.orientation.x
+        q2 = data.pose.pose.orientation.y
+        q3 = data.pose.pose.orientation.z
+        yaw = math.atan2(2.0*(q0*q3 + q1*q2),(1.0-2.0*(q2*q2 + q3*q3)))
+        self.my_pose.theta = yaw
+        self.move()
+        #rospy.loginfo("My goal is %d, my state is %s",self.goal_num,self.state)
+
+    def move(self):
+        if self.state==TURNING:
+            self.turning()
+        elif self.state==STOP_TURNING:
+            self.stop_turning()
+        elif self.state==FORWARD:
+            self.forward()
+        elif self.state==STOP_FORWARD:
+            self.stop_forward()
+
+    
+    def turning(self):
+
+        if (self.reachedAngle()):
+            self.stop()
+            self.state=STOP_TURNING
+            
+        else:
+            aux = Twist()
+            if (abs(self.my_angleGoal-self.my_pose.theta)>3.14):
```

Comment:
Don't hardcode `3.14`. Use the constant from `math` library.

Reply:
Adressed